### ISSUE TITLE
In examples 'PB11' was not declared in this scope

### ISSUE THREAD

**Issue by skruglewicz** [stm32duino/LSM6DSL#15]
Hello, 
I having a problem running all your library examples on my Arduino Uno.
All the code examples error on the Arduino UNO? I get the following compile errors that PB11 is undefined? 
on this section of code?

 // Includes.
#include <LSM6DSLSensor.h>
#define SerialPort Serial
#define I2C2_SCL    PB10
#define I2C2_SDA    PB11
// Components.
**TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);**

Where is PB11 defined? also suggested alternative: 'B11' does not work as well. 
What am I doing wrong? 

Thanks in Advance
Steve K

TERMINAL OUTPUT________________________________________________________________________
Arduino: 1.8.13 (Windows 10), Board: "Arduino Uno"

C:\Program Files (x86)\Arduino\arduino-builder -dump-prefs -logger=machine -hardware C:\Program Files (x86)\Arduino\hardware -hardware C:\Users\skrug\AppData\Local\Arduino15\packages -tools C:\Program Files (x86)\Arduino\tools-builder -tools C:\Program Files (x86)\Arduino\hardware\tools\avr -tools C:\Users\skrug\AppData\Local\Arduino15\packages -built-in-libraries C:\Program Files (x86)\Arduino\libraries -libraries C:\Users\skrug\Documents\Arduino\libraries -fqbn=arduino:avr:uno -vid-pid=0000_0000 -ide-version=10813 -build-path C:\Users\skrug\AppData\Local\Temp\arduino_build_491653 -warnings=default -build-cache C:\Users\skrug\AppData\Local\Temp\arduino_cache_63438 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.arduinoOTA.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\arduinoOTA\1.3.0 -prefs=runtime.tools.arduinoOTA-1.3.0.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\arduinoOTA\1.3.0 -prefs=runtime.tools.avrdude.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino17 -prefs=runtime.tools.avrdude-6.3.0-arduino17.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino17 -prefs=runtime.tools.avr-gcc.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7 -prefs=runtime.tools.avr-gcc-7.3.0-atmel3.6.1-arduino7.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7 -verbose C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino

C:\Program Files (x86)\Arduino\arduino-builder -compile -logger=machine -hardware C:\Program Files (x86)\Arduino\hardware -hardware C:\Users\skrug\AppData\Local\Arduino15\packages -tools C:\Program Files (x86)\Arduino\tools-builder -tools C:\Program Files (x86)\Arduino\hardware\tools\avr -tools C:\Users\skrug\AppData\Local\Arduino15\packages -built-in-libraries C:\Program Files (x86)\Arduino\libraries -libraries C:\Users\skrug\Documents\Arduino\libraries -fqbn=arduino:avr:uno -vid-pid=0000_0000 -ide-version=10813 -build-path C:\Users\skrug\AppData\Local\Temp\arduino_build_491653 -warnings=default -build-cache C:\Users\skrug\AppData\Local\Temp\arduino_cache_63438 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.arduinoOTA.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\arduinoOTA\1.3.0 -prefs=runtime.tools.arduinoOTA-1.3.0.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\arduinoOTA\1.3.0 -prefs=runtime.tools.avrdude.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino17 -prefs=runtime.tools.avrdude-6.3.0-arduino17.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino17 -prefs=runtime.tools.avr-gcc.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7 -prefs=runtime.tools.avr-gcc-7.3.0-atmel3.6.1-arduino7.path=C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7 -verbose C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino

Using board 'uno' from platform in folder: C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.5

Using core 'arduino' from platform in folder: C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.5

Detecting libraries used...

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

Alternatives for LSM6DSLSensor.h: [LSM6DSL-main@2.0.0]

ResolveLibrary(LSM6DSLSensor.h)

  -> candidates: [LSM6DSL-main@2.0.0]

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

Alternatives for Wire.h: [Wire@1.0]

ResolveLibrary(Wire.h)

  -> candidates: [Wire@1.0]

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

Alternatives for SPI.h: [SPI@1.0]

ResolveLibrary(SPI.h)

  -> candidates: [SPI@1.0]

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src\\LSM6DSLSensor.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src\\LSM6DSL_ACC_GYRO_Driver.c" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src\\Wire.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src\\utility\\twi.c" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src\\SPI.cpp" -o nul -DARDUINO_LIB_DISCOVERY_PHASE

Generating function prototypes...

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp" -o "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\preproc\\ctags_target_for_gcc_minus_e.cpp" -DARDUINO_LIB_DISCOVERY_PHASE

"C:\\Program Files (x86)\\Arduino\\tools-builder\\ctags\\5.8-arduino11/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\preproc\\ctags_target_for_gcc_minus_e.cpp"

Compiling sketch...

"C:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++11 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10813 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\cores\\arduino" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\variants\\standard" "-IC:\\Users\\skrug\\Documents\\Arduino\\libraries\\LSM6DSL-main\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\Wire\\src" "-IC:\\Users\\skrug\\AppData\\Local\\Arduino15\\packages\\arduino\\hardware\\avr\\1.8.5\\libraries\\SPI\\src" "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp" -o "C:\\Users\\skrug\\AppData\\Local\\Temp\\arduino_build_491653\\sketch\\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino.cpp.o"

DISCO_IOT_LSM6DSL_DataLog_Terminal:47:21: error: 'PB11' was not declared in this scope

 #define I2C2_SDA    PB11

                     ^

C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino:50:17: note: in expansion of macro 'I2C2_SDA'

 TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);

                 ^~~~~~~~

C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino:47:21: note: suggested alternative: 'B11'

 #define I2C2_SDA    PB11

                     ^

C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino:50:17: note: in expansion of macro 'I2C2_SDA'

 TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);

                 ^~~~~~~~

DISCO_IOT_LSM6DSL_DataLog_Terminal:46:21: error: 'PB10' was not declared in this scope

 #define I2C2_SCL    PB10

                     ^

C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino:50:27: note: in expansion of macro 'I2C2_SCL'

 TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);

                           ^~~~~~~~

C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino:46:21: note: suggested alternative: 'B10'

 #define I2C2_SCL    PB10

                     ^

C:\Users\skrug\AppData\Local\Temp\arduino_modified_sketch_306441\DISCO_IOT_LSM6DSL_DataLog_Terminal.ino:50:27: note: in expansion of macro 'I2C2_SCL'

 TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);

                           ^~~~~~~~

Using library LSM6DSL-main at version 2.0.0 in folder: C:\Users\skrug\Documents\Arduino\libraries\LSM6DSL-main 

Using library Wire at version 1.0 in folder: C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.5\libraries\Wire 

Using library SPI at version 1.0 in folder: C:\Users\skrug\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.5\libraries\SPI 

exit status 1

'PB11' was not declared in this scope

**Comment by fpistm** [comment]
PB11 is STM32 naming. Simply replace those name by Arduino Uno I2C pins (usually D14/D15)

**Comment by skruglewicz** [comment]
I tried several pin names from the following Uno pin diagram snippet and still could not compile.
The pins on the Arduino are
PC5 19 A5 SCL
PC4 18 A4 SDA
<img width="152" alt="image" src="https://user-images.githubusercontent.com/34558711/199383044-87e94e5c-93f5-43f8-87f8-0c0a4b1227f7.png">

_ tried all these using the constants and int values and I'm still getting compile errors.
Here is the screenshot of using the A4 and A5 constants?

<img width="777" alt="image" src="https://user-images.githubusercontent.com/34558711/199387039-281b476c-a582-4f18-9f57-6b6ad2f14197.png">

What type is the constructor using? Is there another way to specify the SDA and SDL pins for the Arduino Uno?
Thanks in Advance
Steve K

**Comment by cparata** [comment]
Hello @skruglewicz ,
as explained by @fpistm , the define in these examples are STM32 specific. For the Arduino UNO you can use directly the Wire instance and connecting the LSM6DSL I2C pins to D14 and D15.
So you can simply try something like:

```
#define dev_i2c Wire

LSM6DSLSensor AccGyr(&dev_i2c, LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH);
```

Please, check how SA0 pin of your LMS6DSL board is connected in your hardware setup. If it is connected to Vdd, you must use LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH as I2C address; if it is connected to GND, you must use LSM6DSL_ACC_GYRO_I2C_ADDRESS_LOW as I2C address.
Best Regards,
Carlo

**Comment by skruglewicz** [comment]
Thank you @cparata
That worked !!! Now I can compile with your suggestion
code as Follows
// Components.
//TwoWire dev_i2c(I2C2_SDA, I2C2_SCL); //not used for Uno only STM32
#include <Wire.h>
#define dev_i2c Wire
LSM6DSLSensor AccGyr(&dev_i2c, LSM6DSL_ACC_GYRO_I2C_ADDRESS_LOW);
//LSM6DSLSensor AccGyr(&dev_i2c, LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH);

I'm not sure what I'm running under as you stated 
"Please, check how SA0 pin of your LMS6DSL board is connected in your hardware setup. If it is connected to Vdd, you must use LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH as I2C address; if it is connected to GND, you must use LSM6DSL_ACC_GYRO_I2C_ADDRESS_LOW as I2C address."
I compiled using both .

Now when I try to execute the example DISCO_IOT_LSM6DSL_DataLogTerminal.ino on my Uno i get zero values
as follows?
16:35:35.580 -> Acc[mg]: 0 0 0 | Gyr[mdps]: 0 0 0
16:35:36.088 -> Acc[mg]: 0 0 0 | Gyr[mdps]: 0 0 0
16:35:36.602 -> Acc[mg]: 0 0 0 | Gyr[mdps]: 0 0 0
16:35:37.115 -> Acc[mg]: 0 0 0 | Gyr[mdps]: 0 0 0

I'm wondering if the I2c is correct? I have several I2C devices   I attached the LSM6DSL CLICK to the MikroBUS #2 on the MikroE Arduino Uno Click.



![image](https://user-images.githubusercontent.com/34558711/199622474-016e1b58-820f-45ac-9548-b8bd3934c1b7.png)

The following picture shows my stacked shilds on my Uno and the MikroE heart rate click attached. 

<img width="314" alt="image" src="https://user-images.githubusercontent.com/34558711/199625556-f6b7f52e-84b1-4d38-bc07-866a66a19cba.png">

<html>
<body>
<p style="color: rgb(0, 0, 0); font-family: Arial, Helvetica, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span><span> </span>I ran my simple i2c_scanner sketch to determine the address of the LSM6DSL.  And sure enough, it displays a new address at 0x70 "17:37:34.015 -&gt; I2C device found at address 0x70  !".</span></p>

{tabbedtable}Without the LSM6DSL attached | With the LSM6DSL Attached
-- | --
7:36:17.551 -> Scanning...17:36:17.596 -> I2C device found at address 0x03 !17:36:17.596 -> I2C device found at address 0x3E !17:36:17.643 -> I2C device found at address 0x57 !17:36:17.690 -> I2C device found at address 0x62 !17:36:17.737 -> done | 17:37:33.823 -> Scanning...17:37:33.870 -> I2C device found at address 0x03 !17:37:33.870 -> I2C device found at address 0x3E !17:37:33.918 -> I2C device found at address 0x57 !17:37:33.966 -> I2C device found at address 0x62 !17:37:34.015 -> I2C device found at address 0x70 !17:37:34.061 -> done

<br class="Apple-interchange-newline">
</body>
</html>

I'm wondering if the library is using the correct device address? 
Thanks in Advance 
Steve K

**Comment by cparata** [comment]
Hello @skruglewicz ,
I checked the schematics of the LSM6DSL CLICK board [here](https://download.mikroe.com/documents/add-on-boards/click/lsm6dsl/lsm6dsl-click-schematic-v100.pdf). You must configure the jumpers of the LSM6DSL CLICK board in I2C mode as explained [here](https://www.mikroe.com/lsm6dsl-click) in the JUMPERS AND SETTINGS section, so you must put:

JP1 -> RIGHT
JP2 -> RIGHT
JP3 -> RIGHT
JP5 -> RIGHT

Finally with JP8 you can decide the I2C slave address of the LSM6DSL:
if JP8 is LEFT you must use LSM6DSL_ACC_GYRO_I2C_ADDRESS_LOW in the code, if JP8 is RIGHT you must use LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH in the code.

The I2C slave address of LSM6DSL should be 0xD4 when you use LSM6DSL_ACC_GYRO_I2C_ADDRESS_LOW and 0xD6 when you use LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH.

Best Regards,
Carlo

**Comment by skruglewicz** [comment]
Thank you @cparata,
Your suggestion to move the jumpers right to select i2c makes sense, however I have never experienced these kinds of jumpers before and I don't understand how to move them right? 
The jumpers  default setting for the click seem to be set for the SPI interface. I'll try and see what it brings?
Thank you for all your help
Steve K

**Comment by cparata** [comment]
Hello @skruglewicz ,
they are solder bridges. So, basically, you must desolder the RIGHT connection and solder the LEFT connection using a coil of solder like in [this video](https://www.youtube.com/watch?v=8JM4oCpWnjU).
If you want to use SPI, I checked the schematics of the Arduino UNO Click Shield [here](https://download.mikroe.com/documents/add-on-boards/click-shields/arduino-uno/arduino-uno-click-shield-schematic-v101.pdf) and it should work. You need only to check what is the correct CS pin according the slot of the Arduino UNO Click Shield that you use for the LSM6DSL CLICK board. The CS pin can be pin 10 or pin 9.

So, instead of:

```
#define dev_i2c Wire

LSM6DSLSensor AccGyr(&dev_i2c, LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH);
```

You can try something like:

```
#define dev_i2c SPI

LSM6DSLSensor AccGyr(&dev_i2c, 10); // Or try 9 instead of 10 according the slot that you are using
```

Best Regards,
Carlo

**Comment by skruglewicz** [comment]
Thank you @cparata for all your great help. 
I was able to figure out how to use the library with the SPI interface on my own.
I defined the SPI object as you described and pasted the CS pin value of 9 (described on the MikroE LSM6DSL product page) to create the instance of the SPI object.
Your suggested code snippit reassured me that I am on the correct path.
Now I need to figure out what to do with the values and experiment more.
Thanks again
Steve K
